soc/riscv/esp32c3/soc.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Include esp-idf headers first to avoid redefining BIT() macro */
+#include <soc/rtc_cntl_reg.h>
+#include <soc/timer_group_reg.h>
+#include <soc/gpio_reg.h>
+#include <soc/syscon_reg.h>
+#include <soc/system_reg.h>
+#include <soc/cache_memory.h>
+#include "hal/soc_ll.h"
+#include "esp_spi_flash.h"
+#include <riscv/interrupt.h>
+
+#include <kernel_structs.h>
+#include <string.h>
+#include <toolchain/gcc.h>
+#include <soc.h>
+
+#define ESP32C3_INTC_DEFAULT_PRIO 15
+
+extern void z_cstart(void);
+extern void esprv_intc_int_set_threshold(int priority_threshold);
+
+/*
+ * This is written in C rather than assembly since, during the port bring up,
+ * Zephyr is being booted by the Espressif bootloader.  With it, the C stack
+ * is already set up.
+ */
+void __attribute__((section(".iram1"))) __start(void)
+{
+		volatile uint32_t *wdt_rtc_protect = (uint32_t *)RTC_CNTL_WDTWPROTECT_REG;
+		volatile uint32_t *wdt_rtc_reg = (uint32_t *)RTC_CNTL_WDTCONFIG0_REG;
+		extern uint32_t _bss_start;
+		extern uint32_t _bss_end;
+
+		/* Configure the global pointer register
+		 * (This should be the first thing startup does, as any other piece of code could be
+		 * relaxed by the linker to access something relative to __global_pointer$)
+		 */
+		__asm__ __volatile__(".option push\n"
+							".option norelax\n"
+							"la gp, __global_pointer$\n"
+							".option pop");
+
+		__asm__ __volatile__("la t0, _esp32c3_vector_table\n"
+							"csrw mtvec, t0\n");
+
+		/* Disable normal interrupts. */
+		csr_read_clear(mstatus, MSTATUS_MIE);
+
+		/* Zero out BSS.  Clobber _bss_start to avoid memset() elision. */
+		(void)memset(&_bss_start, 0, (&_bss_end - &_bss_start) * sizeof(_bss_start));
+		__asm__ __volatile__("" : : "g"(&_bss_start) : "memory");
+
```

Comment:
z_bss_zero() is not suitable?

Reply:
Sure, this was also inherited from esp32 startup code, I added the z_bss_zero in place.